runelite-client/src/main/java/net/runelite/client/plugins/timetracking/farming/CompostTracker.java

```diff
@@ -0,0 +1,264 @@
+	}
+
+	private static final Duration COMPOST_ACTION_TIMEOUT = Duration.ofSeconds(30);
+
+	private static final Pattern COMPOST_USED_ON_PATCH = Pattern.compile(
+		"You treat the (\\w+)( patch)? with (?<compostType>ultra|super|)compost\\.");
+	private static final Pattern FERTILE_SOIL_CAST = Pattern.compile(
+		"The (\\w+)( patch)? has been treated with (?<compostType>ultra|super|)compost\\.");
+	private static final Pattern ALREADY_TREATED = Pattern.compile(
+		"This (\\w+)( patch)? has already been (treated|fertilised) with (?<compostType>ultra|super|)compost( - the spell can't make it any more fertile)?\\.");
+	private static final Pattern INSPECT_PATCH = Pattern.compile(
+		"This is an? (\\w+)( patch)?\\. The soil has been treated with (?<compostType>ultra|super|)compost\\..*");
+
+	private static final ImmutableSet<Integer> compostItems = ImmutableSet.of(
+		ItemID.COMPOST,
+		ItemID.SUPERCOMPOST,
+		ItemID.ULTRACOMPOST,
+		ItemID.BOTTOMLESS_COMPOST_BUCKET_22997
+	);
+
+	private final Client client;
+	private final FarmingWorld farmingWorld;
+	private final ConfigManager configManager;
+
+	@VisibleForTesting
+	final Map<FarmingPatch, PendingCompost> pendingCompostActions = new HashMap<>();
+
+	public void setCompostState(FarmingPatch fp, Compost state)
+	{
+		log.debug("Storing compost state [{}] for patch [{}]", state, fp);
+		if (state == null)
+		{
+			configManager.unsetRSProfileConfiguration(TimeTrackingConfig.CONFIG_GROUP, configKey(fp));
+		}
+		else
+		{
+			configManager.setRSProfileConfiguration(TimeTrackingConfig.CONFIG_GROUP, configKey(fp), state);
+		}
+	}
+
+	public Compost getCompostState(FarmingPatch fp)
+	{
+		return configManager.getRSProfileConfiguration(TimeTrackingConfig.CONFIG_GROUP, configKey(fp), Compost.class);
+	}
+
+	@Subscribe
+	public void onMenuOptionClicked(MenuOptionClicked e)
+	{
+		if (e.isConsumed())
+		{
+			return;
+		}
+
+		Player p = client.getLocalPlayer();
+		Collection<FarmingRegion> currentRegions = farmingWorld.getRegionsForLocation(p.getWorldLocation());
+		if (currentRegions.isEmpty())
+		{
+			return;
+		}
+
+		FarmingPatch farmingPatch = currentRegions.stream()
+			.flatMap(fr -> Arrays.stream(fr.getPatches()))
+			.filter(fp -> fp.getGameObjectIds().contains(e.getId()))
+			.findFirst()
+			.orElse(null);
+		if (farmingPatch == null)
+		{
+			return;
+		}
+		
+		if (e.getMenuAction() == MenuAction.WIDGET_TARGET_ON_GAME_OBJECT)
+		{
+			Widget w = client.getSelectedWidget();
+			assert w != null;
+			int itemId = w.getItemId();
+			if (!compostItems.contains(itemId) && w.getId() != WidgetInfo.SPELL_LUNAR_FERTILE_SOIL.getPackedId())
+			{
+				return;
+			}
+		}
+		else if (!e.getMenuOption().equals("Inspect"))
+		{
+			return;
+		}
+
+		PendingCompost pc = new PendingCompost(
+			Instant.now(),
+			WorldPoint.fromScene(client, e.getParam0(), e.getParam1(), client.getPlane()),
+			farmingPatch
+		);
+		log.debug("Storing pending compost action [{}] for patch [{}]", pc, farmingPatch);
```

Comment:
redundant; there is a debug already when setting config

Reply:
I disagree, this one helps to indicate that we are anticipating a compost action, before the compost is applied.